src/devices/Gpio/Drivers/Sunxi/SunxiDriver.cs

```diff
@@ -0,0 +1,476 @@
+
+            if (unmapped.PortController < 10)
+            {
+                dataPointer = (uint*)(_gpioPointer0 + dataAddress);
+            }
+            else
+            {
+                dataPointer = (uint*)(_gpioPointer1 + dataAddress);
+            }
+
+            uint dataValue = *dataPointer;
+
+            if (value == PinValue.High)
+            {
+                dataValue |= (uint)(1 << unmapped.Port);
+            }
+            else
+            {
+                dataValue &= (uint)~(1 << unmapped.Port);
+            }
+
+            *dataPointer = dataValue;
+        }
+
+        /// <summary>
+        /// Reads the current value of a pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <returns>The value of the pin.</returns>
+        protected unsafe override PinValue Read(int pinNumber)
+        {
+            (int PortController, int Port) unmapped = UnmapPinNumber(pinNumber);
+
+            int dataAddress;
+            uint* dataPointer;
+
+            dataAddress = (CpuxPortBaseAddess + unmapped.PortController * 0x24 + 0x10) & _mapMask;
+
+            if (unmapped.PortController < 10)
+            {
+                dataPointer = (uint*)(_gpioPointer0 + dataAddress);
+            }
+            else
+            {
+                dataPointer = (uint*)(_gpioPointer1 + dataAddress);
+            }
+
+            uint dataValue = *dataPointer;
+
+            return Convert.ToBoolean((dataValue >> unmapped.Port) & 1) ? PinValue.High : PinValue.Low;
+        }
+
+        /// <summary>
+        /// Adds a handler for a pin value changed event.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <param name="eventTypes">The event types to wait for.</param>
+        /// <param name="callback">Delegate that defines the structure for callbacks when a pin value changed event occurs.</param>
+        protected override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventTypes, PinChangeEventHandler callback)
+        {
+            _interruptDriver.OpenPin(pinNumber);
+            _pinModes[pinNumber].InUseByInterruptDriver = true;
+
+            _interruptDriver.AddCallbackForPinValueChangedEvent(pinNumber, eventTypes, callback);
+        }
+
+        /// <summary>
+        /// Removes a handler for a pin value changed event.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <param name="callback">Delegate that defines the structure for callbacks when a pin value changed event occurs.</param>
+        protected override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
+        {
+            _interruptDriver.OpenPin(pinNumber);
```

Comment:
why do we need to open pin in order to remove the callback?

Reply:
I referred to `RaspberryPi3LinuxDriver.cs` before coding😅https://github.com/dotnet/iot/blob/ddeecef97ebf6eb2901b169c46a02af47bd234eb/src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs#L200